Camera2D enhancements (V2)

TwistedTwigleg opened the pull request:
![camera2d_enhancements](https://user-images.githubusercontent.com/25082678/27355809-4ea61c4e-55da-11e7-9677-359e0b1741fa.png)
I added various enhancements for Camera2D:

    Added an option to draw Camera2D's limits
    Added an option to draw Camera2D's drag margin
    Made drawing the Camera's screen bounds optional
    Put all of these option under a new group, "Editor helpers", in Camera2D
    Made Camera2D's editor lines less transparent if is set to current
    Added function calls to _update_scroll() in various places so the editor lines are updated when the values change
    Added variables and set/get functions for toggling the editor lines.

I can remove or change anything if needed. I'm not sure on changing the transparency on the editor lines, but I liked being able to see if a camera was current or not at a glance.

There is one issue I couldn't figure out how to fix. If you scale the Camera2D node by a negative value, the limit bounds editor lines start moving.

(Accidentally messed up git on previous pull request. Made a new pull request to fix the git issues.)

Closes #8175

TwistedTwigleg committed:
Added some editor enhancements to Camera2D. Added the option to draw Camera2Ds bounds and drag margins. Added options to toggle the drawing of Camera2Ds lines in the editor. When a Camera2D is marked as current, its lines become a little less transparent.

TwistedTwigleg commented:
@bojidar-bg I removed the pull request and made this one. Do you mind checking for any issues?

bojidar-bg commented on scene/2d/camera_2d.cpp:
```diff
@@ -267,25 +265,76 @@ void Camera2D::_notification(int p_what) {
-			float area_axis_width = 1;
-			if (current)
-				area_axis_width = 3;
+			if (screen_drawing_enabled == true) {
```
No need to `== true` it, that's not used in core generally.

bojidar-bg commented on scene/2d/camera_2d.cpp:
```diff
@@ -267,25 +265,76 @@ void Camera2D::_notification(int p_what) {
+				Color limit_drawing_color(1, 1, 0, 0.63);
+				float limit_drawing_width = 1;
+				if (current == true) {
+					limit_drawing_color = Color(1, 1, 0, 0.83);
```
You can just do `limit_drawing_color.a = 0.83` here?

TwistedTwigleg commented:
Made changes based on bojidar-bg's review:
- Changed all of the alpha changes from "drawing_color = Color(new values)" to "drawing_color.a = new_alpha value"
- Changed all of the "if (variable == true)" if statements to just "if (variable)"
- Changed all of the if statements checking for current to "if (is_current())" instead of "if (current == true)"


I also squished the commits into one commit.

bojidar-bg reviewed (approved):
LGTM

MarianoGnu commented:
Is it okay to have editor helpers on the inspector? shouldn't it be on editor settings like everithing else?
